Mathlib/CategoryTheory/Limits/Constructions/Over/Products.lean

```diff
@@ -166,15 +166,49 @@ theorem over_hasTerminal (B : C) : HasTerminal (Over B) where
 
 section BinaryProduct
 
-variable {X : C} {Y Z : Over X}
+variable {X : C} (Y Z : Over X)
+
+/--
+Binary fans in the over category is equivalent to pullback cones.
+Also see `CategoryTheory.Over.ConstructProducts.conesEquivFunctor` for the wide pullback version.
+-/
+def pushoutCoconeEquivBinaryCofan : PullbackCone Y.hom Z.hom ≌ BinaryFan Y Z where
+  functor :=
+  { obj c := BinaryFan.mk (Over.homMk (U := Over.mk (c.fst ≫ Y.hom)) (V := Y) c.fst rfl)
+      (Over.homMk (V := Z) c.snd c.condition.symm)
+    map {c1 c2} a := { hom := Over.homMk a.hom, w := by rintro (_|_) <;> aesop_cat } }
+  inverse :=
+  { obj c := PullbackCone.mk c.fst.left c.snd.left (c.fst.w.trans c.snd.w.symm)
+    map {c1 c2} a := { hom := a.hom.left, w := by rintro (_|_|_) <;> simp [← Over.comp_left] } }
+  unitIso := NatIso.ofComponents (fun c ↦ c.eta) (by intros; ext; dsimp; simp)
+  counitIso := NatIso.ofComponents (fun X ↦ Limits.BinaryFan.ext (Over.isoMk (Iso.refl _)
+    (by dsimp; simp)) (by ext; dsimp; simp) (by ext; dsimp; simp))
+    (by intros; ext; dsimp; simp [BinaryFan.ext])
+  functor_unitIso_comp c := by ext; dsimp; simp [BinaryFan.ext]
+
+/-- Binary products in the over category are given by pullbacks. -/
+-- `IsLimit.ofConeEquiv` isn't used here because it gives worse defeqs.
```

Comment:
Can you elaborate on this?

Reply:
It inserts the unit in the equivalence, which makes the lift `id >> pullback.lift` instead.

Reply:
Ah, annoying, let's mention that in this sentence though